integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitRelationalScanIntegrationTestBase.java

```diff
@@ -0,0 +1,210 @@
+import org.junit.jupiter.api.Test;
+
+public abstract class ConsensusCommitRelationalScanIntegrationTestBase
+    extends DistributedTransactionRelationalScanIntegrationTestBase {
+  private DistributedTransactionManager managerWithIncludeMetadataEnabled;
+
+  @BeforeAll
+  @Override
+  public void beforeAll() throws Exception {
+    super.beforeAll();
+
+    Properties includeMetadataEnabledProperties = getPropsWithIncludeMetadataEnabled(getTestName());
+    managerWithIncludeMetadataEnabled =
+        TransactionFactory.create(includeMetadataEnabledProperties).getTransactionManager();
+  }
+
+  @AfterAll
+  @Override
+  public void afterAll() throws Exception {
+    super.afterAll();
+
+    managerWithIncludeMetadataEnabled.close();
+  }
+
+  @Override
+  protected String getTestName() {
+    return "tx_cc";
+  }
+
+  @Override
+  protected final Properties getProperties(String testName) {
+    Properties properties = new Properties();
+    properties.putAll(getProps(testName));
+    if (!properties.containsKey(DatabaseConfig.TRANSACTION_MANAGER)) {
+      properties.setProperty(DatabaseConfig.TRANSACTION_MANAGER, "consensus-commit");
+
+      // Add testName as a coordinator namespace suffix
+      String coordinatorNamespace =
+          properties.getProperty(
+              ConsensusCommitConfig.COORDINATOR_NAMESPACE, Coordinator.NAMESPACE);
+      properties.setProperty(
+          ConsensusCommitConfig.COORDINATOR_NAMESPACE, coordinatorNamespace + "_" + testName);
+    }
+    return properties;
+  }
+
+  protected abstract Properties getProps(String testName);
+
+  protected Properties getPropsWithIncludeMetadataEnabled(String testName) {
+    Properties properties = getProperties(testName);
+    properties.setProperty(ConsensusCommitConfig.INCLUDE_METADATA_ENABLED, "true");
+    return properties;
+  }
+
+  @Test
+  public void scan_PutAndOverlappedRelationalScanGiven_ShouldThrowException()
+      throws TransactionException {
+    // Arrange
+    populateRecords();
+    DistributedTransaction transaction = manager.start();
+    Put put = preparePut(10, NUM_TYPES + 1);
+    Scan scan =
+        Scan.newBuilder()
+            .namespace(namespace)
+            .table(TABLE)
+            .all()
+            .where(ConditionBuilder.column(ACCOUNT_ID).isLessThanOrEqualToInt(10))
+            .consistency(Consistency.LINEARIZABLE)
+            .build();
+
+    // Act
+    Throwable thrown =
+        catchThrowable(
+            () -> {
+              transaction.put(put);
+              transaction.scan(scan);
+              transaction.commit();
+            });
+
+    // Assert
+    assertThat(thrown).isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void scan_PutResultNonOverlappedWithRelationalScanGiven_ShouldThrowException()
+      throws TransactionException {
+    // Arrange
+    populateRecords();
+    DistributedTransaction transaction = manager.start();
+    Put put = preparePut(10, NUM_TYPES);
```

Comment:
Let me confirm this a little bit. If `preparePut` uses the account type as a clustering key as well like `{key: (10, 4), value: 4}`,   the new record wouldn't match any case of https://github.com/scalar-labs/scalardb/blob/eca907801a6561a1eb8dac9dcd92a8ef383463b4/core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java#L279-L281 ?

Reply:
Thank you for the question. Actually, the table schema in `ConsensusCommitRelationalScanIntegrationTestBase` is a little different from `ConsensusCommitIntegrationTestBase`. All account IDs are unique, and there are no clustering keys, i.e., the account type is just a regular column. So, the above test covers case 2, the previous covers case 1, and the next covers case 3.

Reply:
Got it! Thank you!